### PR TITLE
CSS Cleanup and isolation

### LINKED ISSUE #45: RFC: Use SCSS modules
I would like to hear opinions about switching to SCSS modules.
They're exactly like CSS ones just written in SCSS.
I personally think they would allow for much cleaner styling.

Let me know what you think!

CC: @strdr4605

### PR DESCRIPTION
**Pull request by zvonimirr**
As per #45, configured PostCSS to use nesting, sorting, autoprefixing and minifying of CSS.
Also all components are now pretty much isolated which drastically improves DX as we can change CSS without fear
of breaking something important.

Closes #45

### CODE REVIEW

File: styles/FavoritePin.module.css [hunk 1]
```diff
@@ -0,0 +1,26 @@
+.pin {
+  will-change: opacity;
+  opacity: 0;
+  text-decoration: none;
+
+  position: absolute;
+  left: -2rem;
```
Comment (strdr4605) [inline styles/FavoritePin.module.css:7]: Why `-2rem`? isn't `-1.5rem` suitable for most cases?

**Review by strdr4605 — CHANGES_REQUESTED** [review]
Good job.
----
I had some issues running your branch

image

In order to make it work I had to install latest version of 

```diff
@@ -14,9 +14,9 @@
   },
   "dependencies": {
     "favecon": "^1.0.2",
-    "next": "12.1.0",
-    "react": "17.0.2",
-    "react-dom": "17.0.2",
+    "next": "^13.1.6",
+    "react": "^18.2.0",
+    "react-dom": "^18.2.0",
     "rss-parser": "^3.12.0"
   },
   "devDependencies": {
```

**Comment by strdr4605** [comment]
Also wondering if we should keep css files near tsx files. I tend to use this structure:

```
components/
  Component/
    Component.tsx
    Component.module.css
    index.ts // exports Component.tsx
```

But to not make a lot of refactoring, maybe we can experiment with a flat structure for this project:

```diff
components/
  Component1.tsx
  Component1.module.css
  Component2.tsx
  Component2.module.css
pages/
  feed.tsx
  feed.module.css
  index.tsx
- styles/
+ global.css
```

Although I am ok with merging this PR with current folder structure 😊

File: components/ExternalLink.tsx [hunk 2]
```diff
@@ -8,8 +9,13 @@ export interface Props {
 
 export function ExternalLink({ title, link, onClick }: Props) {
   return (
-    // eslint-disable-next-line react/jsx-no-target-blank
-    <a href={link} rel="noopener" target="_blank" onClick={onClick}>
+    <a
+      className={Styles.link}
+      href={link}
+      rel="noopener noreferrer"
```
[the remark this answers is not in the document]
Comment (zvonimirr) [reply components/ExternalLink.tsx:15]: Sure

File: styles/Header.module.css [hunk 3]
```diff
@@ -0,0 +1,49 @@
+.header {
+  max-width: 60ch;
+
+  h1 {
+    cursor: pointer;
+    user-select: none;
+
+    span {
+      color: #ee802f;
```
[the remark this answers is not in the document]
Comment (zvonimirr) [reply styles/Header.module.css:9]: Works for me

File: styles/FavoritePin.module.css (hunk 1 above)
Comment (zvonimirr) [reply styles/FavoritePin.module.css:7]: It actually looks better

**Comment by zvonimirr** [comment]
Yeah I'm pretty much okay with changing the structure but I didn't want to do it in this PR.

**Review by strdr4605 — CHANGES_REQUESTED** [review]
After running `yarn build` on your branch I get ts and js errors

```diff
M components/ExternalLink.tsx
@@ -9,6 +9,7 @@ export interface Props {
 
 export function ExternalLink({ title, link, onClick }: Props) {
   return (
+    // eslint-disable-next-line react/jsx-no-target-blank
     <a
       className={Styles.link}
       href={link}
M pages/feeds.tsx
@@ -14,14 +14,8 @@ interface Props {
   list: JSONFeed[];
 }
 
-interface DialogElement extends HTMLElement {
-  showModal: () => void;
-  close: () => void;
-  open: boolean;
-}
-
 export default function Feeds({ list }: Props) {
-  const dialogRef = useRef<DialogElement>(null);
+  const dialogRef = useRef<HTMLDialogElement>(null);
   const [addedFeed, setAddedFeed] = useState<string>("");
 
   const onAddButtonClick = useCallback(
M pages/index.tsx
@@ -232,7 +232,7 @@ export default function Home() {
                   <>
                     {feed.favicon && (
                       <Image
-                        alt={feed.title}
+                        alt={feed.title || ""}
                         src={feed.favicon}
                         width={16}
                         height={16}
```

---

Added PR check
- https://github.com/cofeed-19/cofeed-19.github.io/commit/f5e34d03b69710e29234f15a2d8c61b01d7266af

so try rebasing on [cofeed-19:master](https://github.com/cofeed-19/cofeed-19.github.io)

**Review by strdr4605 — APPROVED** [review]

**Pull request merged**